### DISCUSSION TITLE
Compiling on MacOS

### DISCUSSION THREAD

**Discussion by kurmasz** [mortbopet/Ripes#100] [Q&A] [answered]
I'm trying to build Ripes on MacOS.  Does this error look familiar to anybody:

$ /Applications/CLion.app/Contents/bin/cmake/mac/bin/cmake .
CMake Error at external/CMakeLists.txt:11 (add_subdirectory):
  The source directory

    /Users/fred/Documents/Code/Ripes/external/VSRTL

  does not contain a CMakeLists.txt file.

-- Configuring incomplete, errors occurred!
See also "/Users/fred/Documents/Code/Ripes/CMakeFiles/CMakeOutput.log".
See also "/Users/fred/Documents/Code/Ripes/CMakeFiles/CMakeError.log".

**Comment by mortbopet** [comment, marked as answer]
Make sure that you pulled the submodules of the project: `git clone --recursive https://github.com/mortbopet/Ripes.git` or `git submodule update --init --recursive`

**Reply by kurmasz** [reply]
That was the problem.  Thanks.

**Reply by kurmasz** [reply]
And now I see that the --recursive part was in the directions all along 🤦 .  (Now I can't yell at my students when they do the same thing ☹️)
